Devo v2 dev

namrata-metron opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues

-  Refactored code for clarity and efficiency
-  Added error-checking mechanisms for robustness
-  Optimized database queries for faster execution
-  Improved documentation for ease of understanding and usage


## Description
This pull request introduces optimizations and improvements to the write_to_table_command and write_to_lookup_table_command functionalities. The changes enhance performance, improve error handling, and streamline code for better maintainability.

## Must have
- [x] Tests
- [x] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=32989) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=32989) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=32989) it.</sub>

namrata-metron reviewed (commented):
enhancement changes

jbabazadeh commented on Packs/Devo/Integrations/Devo_v2/Devo_v2.py:
```diff
@@ -823,13 +824,42 @@ def multi_table_query_command(offset, items):
-    records = check_type(demisto.args()["records"], list)
+    tag = demisto.args().get("tag", None)
+    tableName = demisto.args()["tableName"]
+    records_str = demisto.args()["records"]
```
You can use argToList function cause this records argument IsArray base on the .yml file
this function converts a string representation of args to a python list

and then you can remove the parsing and the check if the records its a list

namrata-metron replied:
Actually the records list array consists of every datatypes i.e. list, boolean, integer, etc. So were directly fetching the data from records parameter. Else if we implement your suggestion, the XSOAR platform gives an error like this, "Failed to execute command devo-write-to-table. Error: the JSON object must be str, bytes or bytearray, not list."

jbabazadeh replied:
You need to add the records without enclosing brackets ([]),  as a list of strings separated by commas.

jbabazadeh commented on Packs/Devo/ReleaseNotes/1_3_1.md:
```diff
@@ -0,0 +1,10 @@
+#### Integrations
+
+##### Devo v2
+- Updated the Docker image to: *demisto/devo:1.0.0.87277*.
+
+- **Release Note Update:**
+  - Improved functionality for writing records to Devo tables and lookup tables has been updated.
+  - The Docker image has been updated to version *demisto/devo:1.0.0.86778*.
+
+These updates enhance the capabilities of the Devo integration by introducing improved methods for writing records to both Devo tables and lookup tables. The new functions provide enhanced flexibility and efficiency in managing and sending data to Devo's platform. Additionally, users can benefit from the latest version of the Docker image, ensuring compatibility and stability with the integration ...
```
```suggestion
 - Improved functionality for **devo-write-to-table** and **devo-write-to-lookup-table** commands.
 - Updated the Docker image to: *demisto/devo:1.0.0.87277*.
```

jbabazadeh replied:
also here need to update the docker image- demisto/devo:1.0.0.87725

jbabazadeh reviewed (changes requested):
Hi @namrata-metron 
Thank you for your contribution!
Good work :)
Please see my comments.
Please feel free to reach out to me with any questions
Thanks again

jbabazadeh commented on Packs/Devo/ReleaseNotes/1_3_1.md:
```diff
@@ -0,0 +1,7 @@
+ - Improved functionality for **devo-write-to-table** and **devo-write-to-lookup-table** commands.
+ - Updated the Docker image to: *demisto/devo:1.0.0.87725*.
+
+These updates enhance the capabilities of the Devo integration by introducing improved methods for writing records to both Devo tables and lookup tables. The new functions provide enhanced flexibility and efficiency in managing and sending data to Devo's platform. Additionally, users can benefit from the latest version of the Docker image, ensuring compatibility and stability with the integration ...
```
Kindly exclude this as it does not in our standard release note format.
```suggestion
```

jbabazadeh reviewed (changes requested):
@namrata-metron ,thanks for the changes you made.
 Please see my comment 
 Please add another test cause its still not reach the coverage is needed.
 Also you need to sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=32989)

namrata-metron commented:
![Screenshot from 2024-02-22 11-28-01](https://github.com/demisto/content/assets/157002078/d158c6e4-2029-41b0-8010-0365a575f32f)
This error is encountered when we try to execute `demisto-sdk pre-commit` command, whereas it isn't related to our code. So please guide us regarding the solution for the same.

jbabazadeh commented:
Hi @namrata-metron try to revert the last changes after this commit [0ab5039](https://github.com/demisto/content/pull/32989/commits/0ab503948da14125424a4897b5776637da557320)

Try to run demisto-sdk pre-commit -g

namrata-metron commented:
> Hi @namrata-metron try to revert the last changes after this commit [0ab5039](https://github.com/demisto/content/pull/32989/commits/0ab503948da14125424a4897b5776637da557320)
> 
> Try to run demisto-sdk pre-commit -g

Yes I have reverted the code and also added additional test cases as well.

jbabazadeh commented:
Hi @namrata-metron ,

We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR.  

Also can you please pull the changes from master? (merge from master into your branch)

namrata-metron commented:
Hi @jbabazadeh,
We are available for the demo meeting today, and our time zone is IST.

namrata-metron reviewed (commented):
Reviewed the changes

jbabazadeh commented:
Hi @namrata-metron , 
Please let me know if you're free tomorrow, Tuesday, at 14:00 IST? 
and  share your email, I'll send you an invitation for the call.

namrata-metron commented:
> Hi @namrata-metron , Please let me know if you're free tomorrow, Tuesday, at 14:00 IST? and share your email, I'll send you an invitation for the call.

Thanks for the update! And I'm available tomorrow for the demo meeting. Here is my mail Id: namrata.gachchi@metronlabs.com

namrata-metron commented:
Hi @jbabazadeh,
Can you please confirm regarding the status of the PR, is it ready for the merge?

jbabazadeh reviewed (approved):
Thanks for the demo

jbabazadeh commented:
Hi @namrata-metron ,
I merged the internal PR, and your update is now in the master branch of content repo.
Your update should be available in the marketplace within the next 72 hours.
Again, thank you very much for your contribution!
